src/log.c

```diff
@@ -3,20 +3,44 @@
  *
  * SPDX-License-Identifier: Apache-2.0 OR MIT
  */
+#include <log.h>
+
+#if defined(STUB_LOG_ENABLED)
+
 #include <stdint.h>
 #include <stdarg.h>
+#include <stddef.h>
 
 #include <target/esp_rom_caps.h>
 #include <target/uart.h>
 
 // These functions are defined in the ROM
 extern void ets_install_uart_printf(void);
 extern void ets_printf(const char *fmt, ...);
+extern void ets_install_putc1(void (*p)(char c));
+extern void ets_install_putc2(void (*p)(char c));
+
+#if defined(STUB_LIB_LOG_BUF)
+struct stub_lib_log_buf g_stub_lib_log_buf;
+static void log_buf_write(char c)
```

Comment:
Naming suggestion for all static functions:  we can add `_internal` suffix like
`stub_lib_log_buf_write_internal`

Reply:
Just my opinion. Agree with @antmak. Maybe we can use shorter suffix? I suppose the main reason for this is to improve readability of the code, to distinguish local functions easily. But long suffixes can do opposite thing. @erhankur @antmak How about `_i` suffix :-)?

Reply:
Okay, my point about static local functions: 1) still prefixed with stub_lib_ (for better searching); 2) no prefix/suffix like `s_` or `_internal` (unlike global or static variables, such suffixes are redundant for functions)